integration-tests/src/test/java/com/datastax/oss/driver/api/core/cql/ExecutionInfoWarningsIT.java

```diff
@@ -169,24 +165,36 @@ public void should_expose_warnings_on_execution_info() {
                 + "APPLY BATCH",
             Strings.repeat("1", 2 * 1024), Strings.repeat("1", 3 * 1024));
     Statement<?> st = SimpleStatement.builder(String.format(query)).build();
-    ResultSet result = SESSION_RULE.session().execute(st);
+    ResultSet result = sessionRule.session().execute(st);
     ExecutionInfo executionInfo = result.getExecutionInfo();
     assertThat(executionInfo).isNotNull();
     List<String> warnings = executionInfo.getWarnings();
     assertThat(warnings).isNotEmpty();
     // verify the log was generated
-    verify(appender, after(500).times(1)).doAppend(loggingEventCaptor.capture());
+    verify(appender, after(500).atLeast(1)).doAppend(loggingEventCaptor.capture());
```

Comment:
what is a reason why it works in at least once way?

Reply:
I think it is explained in the comment above: "the number of warnings and the specific wording varies with different Cassandra versions".

Reply:
Yes, that is correct. For Cassandra 2.2 and 3.0, this query generates 2 warnings, both of which are logged. In 3.11, only one warning is generated.